x/dex/keeper/pool.go

```diff
@@ -180,19 +174,14 @@ func (p *Pool) CalcSharesMinted(
 	return sharesMinted
 }
 
-func (p *Pool) Withdraw(sharesToRemove sdk.Int) (outAmount0 sdk.Int, outAmount1 sdk.Int, err error) {
-	totalShares := &p.LowerTick0.TickData.Reserve0AndShares[p.FeeIndex].TotalShares
-	reserves0 := &p.LowerTick0.TickData.Reserve0AndShares[p.FeeIndex].Reserve0
+func (p *Pool) Withdraw(sharesToRemove sdk.Int, totalShares sdk.Int) (outAmount0 sdk.Int, outAmount1 sdk.Int, err error) {
```

Comment:
Let's init the pool with the total number of shares.

Reply:
see comment above